pkg/autodiscovery/providers/config_reader.go

```diff
@@ -67,11 +67,17 @@ var doOnce sync.Once
 // It reads all configs and caches them in memory for 5 minutes.
 // InitConfigFilesReader should be called at agent startup.
 func InitConfigFilesReader(paths []string) {
+	fileCacheExpiration := 5 * time.Minute
+	if config.Datadog.GetBool("autoconf_config_files_poll") {
+		// Removing some time (1s) to avoid races with polling interval
+		fileCacheExpiration = time.Duration(config.Datadog.GetInt("autoconf_config_files_poll_interval")-1) * time.Second
```

Comment:
Expiration happens once every 30s (see second argument to `cache.New`), chances are low that it will hit at just the right time to remove the items before the next poll.

If we're going to re-populate cache every time any, would it make sense to skip the cache and just store the data all the time explicitly? What would be the impact on memory usage?

Reply:
The cache is actually mostly used in the Cluster Agent, not the Agent, for `advanced_id_identifiers`.

I thought they code in `cache` was checking the expiration time, but it's not the case, I'll handle that manually then.

Reply:
There's a _functional_ race, not a technical race, or maybe I should have used a different wording.
The adjustment avoids to skip some polling intervals due to cache duration.

If I set cache expiration to be == ticker interval, the cache may be used if t1B (cache read time) - t0B (ticker time) < t1A (cache store time) - t0A (ticker time).

And that's **very** likely to be the case as the code path on a cache miss is longer as it actually reads the filesystem.
That would mean that every time we do a cache refresh, the next refresh will use cache, basically halving the expected polling interval.

As the Reader is (unfortunately) common, I don't see any other good way to avoid that without refactoring many bits.

Reply:
Ah, OK, that makes sense!  Perhaps expand the code comment to reflect what you've explained in this PR comment?